modules/version-checker/version_checker.go

```diff
@@ -0,0 +1,190 @@
+package version_checker
+
+import (
+	"fmt"
+	"github.com/gruntwork-io/terratest/modules/shell"
+	"github.com/gruntwork-io/terratest/modules/testing"
+	"github.com/stretchr/testify/require"
+	"regexp"
+	"strconv"
+	"strings"
+)
+
+type VersionCheckerBinary int
+
+// List of binaries supported for version checking.
+const (
+	Docker VersionCheckerBinary = iota
+	Terraform
+	Packer
+)
+
+const (
+	// VersionRegexMatcher is a regex used to extract version string from shell command output.
+	VersionRegexMatcher = "\\d+(\\.\\d+)+"
+	// DefaultVersionArg is a default arg to pass in to get version output from shell command.
+	DefaultVersionArg = "--version"
+)
+
+type CheckVersionParams struct {
+	binary          VersionCheckerBinary
+	expectedVersion string
+	workingDir      string
+}
+
+// CheckVersionE checks whether the given binary version is greater or equal
+// to the given expected version.
+func CheckVersionE(
+	t testing.TestingT,
+	params CheckVersionParams) error {
+
+	err := validateParams(params)
+	if err != nil {
+		return err
+	}
+
+	version, err := getVersionWithShellCommand(t, params)
+	if err != nil {
+		return err
+	}
+
+	return validateBinaryVersionGreaterOrEqual(version, params.expectedVersion)
+}
+
+// CheckVersion checks whether the given binary version is greater or equal to the
+// given expected version and fail if it's not.
+func CheckVersion(
+	t testing.TestingT,
+	params CheckVersionParams) {
+	err := CheckVersionE(t, params)
+	require.NoError(t, err)
+}
+
+// Validate whether the given params contains valid data to check version.
+func validateParams(params CheckVersionParams) error {
+	// Check for empty parameters
+	if len(params.expectedVersion) == 0 {
+		return fmt.Errorf("empty expectedVersion found")
```

Comment:
NIT: enhance this error message to explain to the user what they need to do to remediate their error.

Reply:
Same comment on the next two instances of `fmt.Errorf` in this block.

Reply:
Good idea. Updated.